torch/ao/quantization/pt2e/port_metadata_pass.py

```diff
@@ -88,12 +88,11 @@ def _port_metadata_for_input_quant_nodes(
         scale_node = choose_qparam_users.pop()
         dynamic_q_node = next(iter(scale_node.users.keys()))
         dynamic_q_node_users = _filter_sym_size_users(dynamic_q_node)
-        if len(dynamic_q_node_users) > 1:
```

Comment:
I think maybe we just don't duplicate in this case and have the lowering pass to manually match the chain? something we did in https://github.com/pytorch/pytorch/pull/83459

Reply:
Hi @jerryzh168 Are you suggesting we modify the `convert_pt2e` path and do not duplicate `dq` in the dynamic quant case? Thanks

Reply:
yeah, I feel it's better to either not duplicate or duplicate "choose_qparams -> q -> dq". and not duplicate anything feels cleaner

Reply:
Agree with @jerryzh168. My preference would be to duplicate the whole chain. I would have that when i added duplicate dq, it was already doing that but maybe not.

My rationale for duplicating whole chain is that the definition of say "DQ linear" is "fp32-> chose_qparams -> q -> dq ....". So if you ever pattern match to find DQ linear, you want to match that. But this argument is solely driven from pattern matching perspective.

Reply:
Hi @jerryzh168 @kimishpatel May I know whether you have reached an agreement? Jerry was suggesting not duplicating `choose_qparam -> q -> dq` while Kimish's preference is to duplicate the whole chain. What is the conclusion?

Reply:
@Xia-Weiwen can you say more on why do you have a pattern where dynamic quant node iis consumed by multiple nodes? Are you dynamically quantizing output of a node? And then there are multiple consumers? In this case duplicating the entire chain can result in sub optimal performance in inductor backend? While duplicating only dq still allows you to share "chose qparam -> q"? If so I think what Jerry said would make sense. That is just not duplicate dq at all.

Reply:
Hi @kimishpatel. A typical case is the Q/K/V projections in self-attention. The pattern is like:
```
 linear (Q)
  /
x --- linear (K)
  \
 linear (V)
```
With current implementation of quantization, the pattern becomes the following after the `DuplicateDQPass`:
```
                     dq - linear (Q)
                     /
x - choose_qparam - q - dq - linear (K)
                     \
                     dq - linear (V)
```
This pattern with duplicate dq is what we want for the X86Inductor backend. We plan to lower it by fusing `dq - linear` only as static quantized linear:
```
                     static_quant_linear (Q)
                     /
x - choose_qparam - q - static_quant_linear (K)
                     \
                     static_quant_linear (V)
```
Because we don't want to include `choose_qparam` and `q` in the kernel, and we can reuse the code for static quantization.
As you have agreed, let's not duplicate `dq` at all for dynamic quantization. The X86Inductor backend can duplicate `dq` when doing the lowering. How does that sound to you? Thanks.

Reply:
I have updated this PR. Please review again. Thanks!

Reply:
> As you have agreed, let's not duplicate dq at all for dynamic quantization. The X86Inductor backend can duplicate dq when doing the lowering. How does that sound to you? Thanks.

Yeah that makes sense.

BTW how are you doing annotation for quantization for this pattern? Is it quantizing X, i.e. output of attention layer, via dynamic quantization OR is it doing dynamic quantization of q, k, v linear projections in attention? Former is dynamic quant on output, the latter is on input.

Reply:
@kimishpatel it's probably annotated with this: https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/quantizer/x86_inductor_quantizer.py#L932

basically the partition does not know about connection information so it will just find one linear pattern in
```
x -> linear1
 \ -> linear2
 \ -> linear3
```
and annotate it

Reply:
Hi @jerryzh168 Why did you say it would just find one linear pattern? I think all three linear patterns are found and annotated.

Reply:
@Xia-Weiwen my question is whether you are annotating input activation with dynamic quant config for all 3 linears. If so I would have expected to see 3 chains of chose qparams -> q -> dq.

Reply:
I see. We see three chains of `dq - linear`.

Reply:
> Hi @jerryzh168 Why did you say it would just find one linear pattern? I think all three linear patterns are found and annotated.

oh sorry, I think it will find three patterns and annotated three times as well, but we only insert one observer for input because of the deduplication logic